src/observed/covariance.jl

```diff
@@ -39,11 +39,11 @@ use this if you are sure your covariance matrix is in the right format.
 ## Additional keyword arguments:
 - `spec_colnames::Vector{Symbol} = nothing`: overwrites column names of the specification object
 """
-struct SemObservedCovariance{B, C, D, O} <: SemObserved
+struct SemObservedCovariance{B, C} <: SemObserved
```

Comment:
I'm wondering what best practices is here - I think the old version should not have downsides in terms of performance; restricting the types of these two fields is maybe best done with:
```suggestion
struct SemObservedCovariance{B, C, D <: Int, O <: Int} <: SemObserved
```
I'm not sure when/if annotating the types of specific fields vs. having a parametric type is preferred, maybe you have some thoughts on that?

Reply:
As far as I know, type parameters add some stress on Julia compiler, so if they could be avoided (i.e. if the type is fixed), they should be avoided.
Plus, it makes the code less readable.
In this particular case, it is quite reasonable to fix the number of observations and the number of variables to `Int`.
I don't see usecases, where anything else would be required.

Reply:
Also, `D <: Int` is, sort of, no-op as `Int` is a concrete type, i.e. it just tells that `D === Int`.
`D <: Integer` allows flexibility, since `Integer` is an abstract type, but, as I wrote in the previous comment, I don't see the benefit of this flexibility here.

Reply:
BTW, here's the [performance tips about type declarations](https://docs.julialang.org/en/v1/manual/performance-tips/#Type-declarations) from Julia manual

Reply:
I see, then let's do it like you suggested. In the manual it only says if you have abstract field types it's better to use parametric Types; but here the field types are concrete (so I guess that makes it okay).